tests/test_reactive.py

```diff
@@ -709,6 +709,268 @@ def second_callback() -> None:
         assert logs == ["first", "second", "first", "second"]
 
 
+async def test_watch_decorator():
+    """Test watchers defined via a decorator."""
+
+    class WatchApp(App):
+        count = reactive(0, init=False)
+
+        watcher_call_count = 0
+
+        @count.watch
+        def _(self, value: int) -> None:
+            self.watcher_call_count += 1
+
+    app = WatchApp()
+    async with app.run_test():
+        app.count += 1
+        assert app.watcher_call_count == 1
+        assert app.count == 1
+        app.count += 1
+        assert app.watcher_call_count == 2
+        assert app.count == 2
+        app.count -= 1
+        assert app.watcher_call_count == 3
+        assert app.count == 1
+        app.count -= 1
+        assert app.watcher_call_count == 4
+        assert app.count == 0
+
+
+async def test_watch_decorator_call():
+    """Test watchers defined via a decorator, that is called."""
+
+    class WatchApp(App):
+        count = reactive(0, init=False)
+
+        watcher_call_count = 0
+
+        @count.watch()
```

Comment:
It was _very_ difficult for me to spot the difference.
Would it be worth adding a comment `# <--` to make the diff obvious so that no one deletes this test in the future by mistake?